ansible/roles/debops.postfix/defaults/main.yml

```diff
@@ -256,6 +271,351 @@ postfix__tls_dh512_param_file: '{{ ansible_local.dhparam[postfix__dhparam_set]
                                        ansible_local.dhparam[postfix__dhparam_set]|d())
                                    else "" }}'
                                                                    # ]]]
+# .. envvar:: postfix__no_log [[[
+#
+# This variable can be used in the Ansible tasks to set the ``no_log``
+# parameter. Changing it to ``False`` can help with debugging issues with the
+# roles, but otherwise it should be set to ``True`` to ensure that Ansible
+# doesn't leak sensitive data in its logs.
+postfix__no_log: True
+                                                                   # ]]]
+                                                                   # ]]]
+# Virtual Mail [[[
+# ----------------
+#
+# This settings help to configure to enable `postfix` to host multiple (virtual) domains,
+# and thus provide email to several domains with just one `mail server`.
+# Currently the Virtual Mail support works only with LDAP enabled, in the future `mariaDB` could be enabled.
+#
+
+# .. envvar:: postfix__ldap_enabled [[[
+#
+# In order to enable Virtual Mail support, you need to enable LDAP authentication support manually.
+# It is disabled by default in order to define a single Mail Relay Server.
+# When enabled, `postfix` will configure mail-boxes locally and allow LDAP authenticated users to
+# send email over this instance.
+postfix__ldap_enabled: 'False'
+                                                                   # ]]]
+# .. envvar:: postfix__virtual_mail_posix_user [[[
+#
+# Virtual Mail POSIX username
+# For the accesses to the mailbox directories a separate user `vmail` (`Virtual Mail`) is created,
+# under which the accesses of Postfix, Dovecot and other components of the mail server should take place.
+# On the one hand this prevents mail server components from accessing sensitive system directories,
+# on the other hand it protects the mailboxes from external access.
+# Only vmail (and root) are allowed to access the mailboxes.
+postfix__virtual_mail_posix_user: 'vmail'
+                                                                   # ]]]
+# .. envvar:: postfix__virtual_mail_posix_uidnumber [[[
+#
+# Virtual Mail POSIX uidNumber
+#
+postfix__virtual_mail_posix_uidnumber: 5000
+                                                                   # ]]]
```

Comment:
You shouldn't specify static UID/GID allocations in a role, especially in the normal user range above >1000. Just make sure that the `vmail` UNIX group and account are created with the "system" parameter so that they have UID/GID below <1000.

In a single host setup you don't have to care what specific UID/GID the group and account has. In a distributed setup it might be better to define such account in LDAP itself so that it is consistent across hosts. For this, you probably would want to add a variable to enable/disable local account and group creation so that the relevant Ansible modules are skipped and don't break the run when they notice that the account and group is already present in LDAP.

Also, you shouldn't add an empty line at the end of the comment here, before the actual variable.

Reply:
Without a fixed uid/gid I had problems with the permissions. 
Maybe generate system user and fetch the uid/gid and store it as fact?

Reply:
This could be handled by using the `getent` Ansible module and getting the UID/GID of the specified user and group at runtime... But knowing the exact reason could probably give a better solution.